blaze/compute/core.py

```diff
@@ -544,7 +544,7 @@ def coerce_core(result, dshape, odo_kwargs=None):
         return result
     elif isscalar(dshape):
         result = coerce_scalar(result, dshape, odo_kwargs=odo_kwargs)
-    elif istabular(dshape) and isrecord(dshape.measure):
+    elif istabular(dshape) or isrecord(dshape):
```

Comment:
I don't think this is correct, a scalar record is _not_ a dataframe, it is a single row of a dataframe.

Reply:
Depends on how we want to treat the selection of a single row. If it's a scalar then `isscalar` can be extended to handle it.